docs/04-reference/winglang-spec.md

```diff
@@ -765,15 +767,95 @@ let handler2 = inflight() => {
 }
 ```
 
-Bridge between preflight and inflight is crossed with the help of immutable data
-structures, "structs" (user definable and `Struct`), and the capture mechanism.
+The bridge between preflight and inflight is crossed with the help of immutable data
+structures, user-defined structs, resources, and the capture mechanism.
 
 Preflight resource methods and initializers can receive an inflight function as
 an argument. This enables resources to define code that will be executed on the
 deployed resource (lambda functions, docker, virtual machines etc).
 
 [`▲ top`][top]
 
+#### 1.3.1 Phase-independent code
+
+Code that is not dependent on the phase of execution can be designated as phase-independent using the `?inflight` modifier.
```

Comment:
What if all functions that don't have an explicit phase modifier are maybe-inflights by default?

Reply:
@skyrpex If I'm understanding correctly, you're proposing one of two alternatives. Either (1) to have the compiler infer that a function is phase-independent where possible, OR (2) to make phase-independent the default for functions, and make preflight-only functions require a keyword (like `preflight () => {}`).

I'll try to explain why I chose my current proposal over those options (will also add to the PR description)

----

For the first alternative, I feel like the phase of a function isn't something we should allow the compiler to infer -- not because it's not technically feasible, but of user experience problems it could introduce.

In the future, devs will be able to create a Wing libraries that export resources with public preflight methods and public inflight methods. When using a Wing library, consumers should be able to rely on the fact that the method always works in that phase in the future, even if it the implementation changes -- just like today you can rely on the fact that a if a imported TypeScript function returns a boolean, then it will always return a boolean in the future (unless there is a breaking change).

If the compiler inferred whether a function should be phase-independent or preflight, then changing the implementation of the function so that it captures some mutable data could change the function's type to preflight, thus causing a potentially breaking change to consumers.

----

For the second alternative, I think the idea of functions being phase-independent as the default could be interesting. But I feel like it might be more confusing since a lot of the Wing language is currently designed around the concept that "the top level of your code is preflight" -- for example, creating a bucket at the top-level scope of a Wing file happens in preflight. And the default constructor of a resource is its preflight constructor, which matches up with that. 

Put another way, I think preflight-only functions will be more common than phase-independent functions, so I think there should be more friction for the phase-independent use case than the preflight code use case. But this is definitely something we can experiment with.